emulators/retroarch/files/patch-retroarch-library-dirs.diff

```diff
@@ -0,0 +1,174 @@
+    SETTING_PATH("thumbnails_directory",
+          settings->paths.directory_thumbnails, true, NULL, true);
+    SETTING_PATH("playlist_directory",
+-         settings->paths.directory_playlist, true, NULL, true);
++         settings->paths.directory_playlist, true, "~/Library/Application Support/RetroArch/playlists", true);
+    SETTING_PATH("content_favorites_directory",
+          settings->paths.directory_content_favorites, true, NULL, true);
+    SETTING_PATH("content_history_directory",
+@@ -1356,26 +1356,26 @@
+    SETTING_PATH("runtime_log_directory",
+          settings->paths.directory_runtime_log, true, NULL, true);
+    SETTING_PATH("joypad_autoconfig_dir",
+-         settings->paths.directory_autoconfig, false, NULL, true);
++         settings->paths.directory_autoconfig, false, "~/Library/Application Support/RetroArch/autoconfig", true);
+    SETTING_PATH("audio_filter_dir",
+-         settings->paths.directory_audio_filter, true, NULL, true);
++         settings->paths.directory_audio_filter, true, "~/Library/Application Support/RetroArch/audio_filters", true);
+    SETTING_PATH("savefile_directory",
+-         dir_get_ptr(RARCH_DIR_SAVEFILE), true, NULL, false);
++         dir_get_ptr(RARCH_DIR_SAVEFILE), true, "~/Library/Application Support/RetroArch/saves", false);
+    SETTING_PATH("savestate_directory",
+-         dir_get_ptr(RARCH_DIR_SAVESTATE), true, NULL, false);
++         dir_get_ptr(RARCH_DIR_SAVESTATE), true, "~/Library/Application Support/RetroArch/states", false);
+ #ifdef HAVE_MENU
+    SETTING_PATH("rgui_browser_directory",
+          settings->paths.directory_menu_content, true, NULL, true);
+    SETTING_PATH("rgui_config_directory",
+-         settings->paths.directory_menu_config, true, NULL, true);
++         settings->paths.directory_menu_config, true, "~/Library/Application Support/RetroArch/config", true);
+ #endif
+ #ifdef HAVE_OVERLAY
+    SETTING_PATH("overlay_directory",
+-         settings->paths.directory_overlay, true, NULL, true);
++         settings->paths.directory_overlay, true, "~/Library/Application Support/RetroArch/overlays", true);
+ #endif
+ #ifdef HAVE_VIDEO_LAYOUT
+    SETTING_PATH("video_layout_directory",
+-         settings->paths.directory_video_layout, true, NULL, true);
++         settings->paths.directory_video_layout, true, "~/Library/Application Support/RetroArch/layouts", true);
+ #endif
+    SETTING_PATH(
+          "screenshot_directory",
+@@ -1392,7 +1392,7 @@
+       }
+    }
+ 
+-   SETTING_ARRAY("log_dir", settings->paths.log_dir, true, NULL, true);
++   SETTING_ARRAY("log_dir", settings->paths.log_dir, true, "~/Library/Application Support/RetroArch/logs", true);
+ 
+    *size = count;
+ 
+@@ -3099,10 +3099,10 @@
+       size_t tmp = 0;
+       if (config_get_size_t(conf, size_settings[i].ident, &tmp))
+          *size_settings[i].ptr = tmp ;
+-      /* Special case for rewind_buffer_size - need to convert 
++      /* Special case for rewind_buffer_size - need to convert
```

Comment:
This and all subsequent changes seem to be just whitespace changes and could be removed since they aren't relevant to the issue being fixed.